### ISSUE TITLE
[release] IceRec 5.1 Release for pass2 (Trac #1924)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1916]
This will serve as a tracking ticket for IceRec V05-01-00, which is was will be used for pass2 reprocessing.  The candidate is currently based on the latest stable combo.

http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC

Migrated from https://code.icecube.wisc.edu/ticket/1924
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:32", 
    "description": "This will serve as a tracking ticket for IceRec V05-01-00, which is was will be used for pass2 reprocessing.  The candidate is currently based on the latest stable combo.\n\nhttp://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC", 
    "reporter": "olivas", 
    "cc": "claudio.kopper, olivas, joertlin", 
    "resolution": "fixed", 
    "_ts": "1550067212810271", 
    "component": "combo reconstruction", 
    "summary": "[release] IceRec 5.1 Release for pass2", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-12-20T15:39:40", 
    "milestone": "", 
    "owner": "kjmeagher", 
    "type": "task"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20161220T15:39:49: 
* olivas changed status from "new" to "accepted"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20161220T17:46:13: olivas commented:

> Notes from the phone call:
> 1) Need the new EHE filter implementation from Ming.
> 2) Decide whether we're going with the latest SPE fits from Spencer, which is naturally included in the candidate since it was cut from the trunk.  This will be discussed with the calibration group (or just Dawn at the very least).

**Comment by icecube-trac** [comment]
Trac update at 20170109T15:41:17: joertlin commented:

> The project `vemcal` is missing

**Comment by icecube-trac** [comment]
Trac update at 20170109T17:37:46: olivas commented:

> vemcal was just added to the latest candidate:
> http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC2

**Comment by icecube-trac** [comment]
Trac update at 20170120T20:02:27: joertlin commented:

> filterscript modifications are required:
> * resources/pass2/process_sdst_archive_to_l2.py
> * python/offlineL2/SpecialWriter.py
>

**Comment by icecube-trac** [comment]
Trac update at 20170221T15:39:53: olivas changed milestone from "December 2016" to "February 2017"

**Comment by icecube-trac** [comment]
Trac update at 20170228T16:40:52: 
* olivas changed status from "accepted" to "assigned"
* olivas changed cc from "claudio.kopper, kjmeagher, joertlin" to "claudio.kopper, olivas, joertlin"
* olivas changed owner from "olivas" to "kjmeagher"
* olivas commented:

> Note: I need to get an offline release out for this.  I'm working on this this week.

**Comment by icecube-trac** [comment]
Trac update at 20170310T20:38:27: olivas commented:

> Replying to [comment:7 olivas]:
> > Note: I need to get an offline release out for this.  I'm working on this this week.
> 
> 
> The offline candidate is out and will be the basis for both IceRec 5.1 (for pass2 and pole) and IceSim 5.1.
> 
> Two new projects that need review for online filters this season is ddddr and StartingTrackVeto (http://code.icecube.wisc.edu/svn/sandbox/kjero/StartingTrackVeto/)

**Comment by icecube-trac** [comment]
Trac update at 20170310T20:58:41: 
* olivas commented:

> Maintainers for the following projects need to ensure their release notes are up to data and accurate.
> 
> Update this doc when completed ​https://goo.gl/UQSsNx
> 
> AtmCscdEnergyReco 
> BadDomList
> dst
> wavereform 
> topeventcleaning
> tpx
> linefit 
> improvedLinefit 
> dipolefit
> tensor-of-inertia
> clast
> cscd-llh
> credo
> toprec
> fill-ratio
> portia
> smallshower-filter
> finiteReco
> mue
> ophelia
> static-twc
> coinc-twc 
> cramer-rao
> core-removal
> DeepCore_Filter
> trigger-splitter 
> TopologicalSplitter
> millipede
> truncated_energy
> CascadeVariables
> CommonVariables
> SLOPtools
> KalmanFilter 
> pybdt
> NoiseEngine 
> shield 
> HiveSplitter
> spline-reco
> STTools 
> SeededRTCleaning
> VHESelfVeto 
> CoincSuite 
> IceHive
> full_event_followup 
> frame_object_diff 
> recclasses 
> gulliver
> gulliver-bootstrap 
> gulliver-modules
> lilliput
> paraboloid
> bayesian-priors
> ipdf 
> double-muon
> level3-filter-cascade
> level3-filter-muon
> level3-filter-lowen 
> 
> 


* olivas changed _comment0 from:

> Maintainers for the following projects need to ensure their release notes are up to data and accurate.
> 
> Here's the full list of maintainers: https://goo.gl/InZ0PH
> 
> AtmCscdEnergyReco 
> BadDomList
> dst
> wavereform 
> topeventcleaning
> tpx
> linefit 
> improvedLinefit 
> dipolefit
> tensor-of-inertia
> clast
> cscd-llh
> credo
> toprec
> fill-ratio
> portia
> smallshower-filter
> finiteReco
> mue
> ophelia
> static-twc
> coinc-twc 
> cramer-rao
> core-removal
> DeepCore_Filter
> trigger-splitter 
> TopologicalSplitter
> millipede
> truncated_energy
> CascadeVariables
> CommonVariables
> SLOPtools
> KalmanFilter 
> pybdt
> NoiseEngine 
> shield 
> HiveSplitter
> spline-reco
> STTools 
> SeededRTCleaning
> VHESelfVeto 
> CoincSuite 
> IceHive
> full_event_followup 
> frame_object_diff 
> recclasses 
> gulliver
> gulliver-bootstrap 
> gulliver-modules
> lilliput
> paraboloid
> bayesian-priors
> ipdf 
> double-muon
> level3-filter-cascade
> level3-filter-muon
> level3-filter-lowen 
> 
> 

to:

> 1489181226893228


* olivas changed _comment1 from:

> Maintainers for the following projects need to ensure their release notes are up to data and accurate.
> 
> Here's the full list of maintainers: https://goo.gl/InZ0PH
> 
> Update this doc when completed ​https://goo.gl/UQSsNx
> 
> AtmCscdEnergyReco 
> BadDomList
> dst
> wavereform 
> topeventcleaning
> tpx
> linefit 
> improvedLinefit 
> dipolefit
> tensor-of-inertia
> clast
> cscd-llh
> credo
> toprec
> fill-ratio
> portia
> smallshower-filter
> finiteReco
> mue
> ophelia
> static-twc
> coinc-twc 
> cramer-rao
> core-removal
> DeepCore_Filter
> trigger-splitter 
> TopologicalSplitter
> millipede
> truncated_energy
> CascadeVariables
> CommonVariables
> SLOPtools
> KalmanFilter 
> pybdt
> NoiseEngine 
> shield 
> HiveSplitter
> spline-reco
> STTools 
> SeededRTCleaning
> VHESelfVeto 
> CoincSuite 
> IceHive
> full_event_followup 
> frame_object_diff 
> recclasses 
> gulliver
> gulliver-bootstrap 
> gulliver-modules
> lilliput
> paraboloid
> bayesian-priors
> ipdf 
> double-muon
> level3-filter-cascade
> level3-filter-muon
> level3-filter-lowen 
> 
> 

to:

> 1489181238500427

**Comment by icecube-trac** [comment]
Trac update at 20170310T20:59:53: 
* olivas changed milestone from "February 2017" to "March 2017"
* olivas changed summary from "[release] IceRec Release for pass2" to "[release] IceRec 5.1 Release for pass2"

**Comment by icecube-trac** [comment]
Trac update at 20170313T19:59:36: olivas commented:

> Latest RC is here: http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC5
> 
> I lost track of this, so figured I make a note of it in this tracking ticket.

**Comment by icecube-trac** [comment]
Trac update at 20170313T22:26:31: olivas commented:

> Make and note that we should either resolve this ticket or determine it's not a real problem before releasing: http://code.icecube.wisc.edu/projects/icecube/ticket/1948

**Comment by icecube-trac** [comment]
Trac update at 20170314T16:28:46: joertlin commented:

> I just noticed that SeededRTCleaning is still in the metaproject. It was already removed since it was retired but the Cascade L3 script wanted to use it though. However, I propose to remove SeededRT and modify the Cascade L3 script accordingly.

**Comment by icecube-trac** [comment]
Trac update at 20170314T18:07:16: olivas commented:

> I asked Yiqian to verify that cascade L3 no longer needs this and/or has been updated accordingly.

**Comment by icecube-trac** [comment]
Trac update at 20170315T16:22:38: kjmeagher commented:

> dtosi needs the commit to make it into the release http://code.icecube.wisc.edu/projects/icecube/changeset/153889/IceCube

**Comment by icecube-trac** [comment]
Trac update at 20170321T02:02:36: olivas commented:

> http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC7
> 
> This includes the latest shield, along with recclasses, toprec, and the offline candidate.

**Comment by icecube-trac** [comment]
Trac update at 20170331T19:36:27: olivas commented:

> I really want to clear this one before the next full release.
> 
> http://code.icecube.wisc.edu/projects/icecube/ticket/1894

**Comment by icecube-trac** [comment]
Trac update at 20170331T19:36:43: olivas commented:

> Latest candidate is here: http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC8

**Comment by icecube-trac** [comment]
Trac update at 20170331T20:13:46: olivas commented:

> Removed ddddr from the trunk of both combo and icrec.
> 
> Seems there are dependency issues with Minuit2.
> 
> In file included from
>     /home/olivas/icecube/combo/fresh_trunk/src/ddddr/public/ddddr/I3MuonEnergy.
>     h:35:0,
> from /home/olivas/icecube/combo/fresh_trunk/src/ddddr/private/ddddr/I3MuonEnergy.cxx:11:
>     /home/olivas/icecube/combo/fresh_trunk/src/ddddr/private/ddddr/MuonEnergyMinuit2.h:7:29:
>     fatal error: Minuit2/FCNBase.h: No such file or directory
> compilation terminated.
>

**Comment by icecube-trac** [comment]
Trac update at 20170405T16:02:59: olivas commented:

> Don't forget to add gcdserver to the release candidate.

**Comment by icecube-trac** [comment]
Trac update at 20170405T16:29:35: joertlin commented:

> Also the BadDomList project is needed for L2 offline processing. The BadDomList project needs a new release (just committed required changes to the trunk) since it was using `OMKey.is_icetop()` that has been removed.

**Comment by icecube-trac** [comment]
Trac update at 20170410T18:05:54: 
* joertlin commented:

> We need to add the latest versions of the SPE correction files for pass2 (seasons 2011 - 2014). They are not in the trunk yet but they should be generated and committed today.


* joertlin changed _comment0 from "We need to add the latest SPE correction files to icerec for pass2. They are not in the trunk yet but they should be generated and committed today." to "1491848396298142"

**Comment by icecube-trac** [comment]
Trac update at 20170410T18:08:47: 
* olivas changed _comment2 from:

> I'm going to leave the RC9 (http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC9) slushy for another day.  
> 
> Waiting on:
> * gcdserver
> * SPE corrections
> * StartingTrackVeto test
> * ddddr (more changes in the hopper)
> 
> Aside from filterscripts, is there anything else?

to:

> 1491922569317628


* olivas changed _comment0 from:

> I'm going to leave the RC9 slushy for another day.  Waiting on:
> * gcdserver
> * SPE corrections
> * StartingTrackVeto test
> 
> Aside from filterscripts, is there anything else?

to:

> 1491848273735830


* olivas commented:

> I'm going to leave the RC9 (http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC9) slushy for another day.  
> 
> Waiting on:
> * ~~gcdserver~~ (done)
> * ~~SPE corrections~~ (committed to filterscripts)
> * StartingTrackVeto test (needs safer failure modes beyond throws and seg faults)
> * ~~ddddr (more changes in the hopper)~~ (new release V01-00-01)
> 
> ~~Aside from filterscripts, is there anything else?~~  (filterscripts is tagged)


* olivas changed _comment3 from:

> I'm going to leave the RC9 (http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC9) slushy for another day.  
> 
> Waiting on:
> * ~~gcdserver~~ (done)
> * ~~SPE corrections~~ (committed to filterscripts)
> * StartingTrackVeto test
> * ddddr (more changes in the hopper)
> 
> Aside from filterscripts, is there anything else?

to:

> 1491942517091795


* olivas changed _comment4 from:

> I'm going to leave the RC9 (http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC9) slushy for another day.  
> 
> Waiting on:
> * ~~gcdserver~~ (done)
> * ~~SPE corrections~~ (committed to filterscripts)
> * StartingTrackVeto test
> * ~~ddddr (more changes in the hopper)~~ (new release V01-00-01)
> 
> Aside from filterscripts, is there anything else?

to:

> 1492112420577088


* olivas changed _comment1 from:

> I'm going to leave the RC9 (http://code.icecube.wisc.edu/svn/meta-projects/icerec/candidates/V05-01-00-RC9) slushy for another day.  
> 
> Waiting on:
> * gcdserver
> * SPE corrections
> * StartingTrackVeto test
> 
> Aside from filterscripts, is there anything else?

to:

> 1491848338913029

**Comment by icecube-trac** [comment]
Trac update at 20170410T19:03:09: joertlin commented:

> Pass2 SPE correction files committed (see http://code.icecube.wisc.edu/projects/icecube/changeset/154852/IceCube)

**Comment by icecube-trac** [comment]
Trac update at 20170413T19:40:40: olivas commented:

> Replying to [comment:24 joertlin]:
> > Pass2 SPE correction files committed (see http://code.icecube.wisc.edu/projects/icecube/changeset/154852/IceCube)
> 
> Included in the latest candidate.

**Comment by icecube-trac** [comment]
Trac update at 20170417T19:55:35: joertlin commented:

> Today, I got updated SPE correction files from Spencer. Looks like that they are really final now. http://code.icecube.wisc.edu/projects/icecube/changeset/155046/IceCube

**Comment by icecube-trac** [comment]
Trac update at 20170422T05:39:40: 
* kjmeagher changed status from "assigned" to "closed"
* kjmeagher changed resolution from "" to "fixed"
* kjmeagher commented:

> IceRec V5.1 was released in https://code.icecube.wisc.edu/ticket/155195

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:32: 
* olivas commented:

> Milestone March 2017 deleted


* olivas changed milestone from "March 2017" to ""
